src/Servers/Kestrel/Core/src/Internal/Http/HttpRequestPipeReader.cs

```diff
@@ -80,6 +80,7 @@ public void StartAcceptingReads(MessageBody body)
             {
                 _state = HttpStreamState.Open;
                 _body = body;
+                _error = null;
```

Comment:
The problem I have with this fix is that it implies that something is aborting the stream after we pool it given the following check.

https://github.com/dotnet/aspnetcore/blob/39912f8f73e06f5d9735745d24545aed8821698f/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs#L169

If we're pooling streams when they're still referenced and can still be aborted, this is likely just making the race less likely rather than fixing it.

> This triggers the internal error, though I don't know why yet, but maybe ExceptionDispatchInfo.Throw() shouldn't be called twice on the same instance.

This should be fine. It's super common.